tidy3d/components/data/monitor_data.py

```diff
@@ -1081,6 +1081,7 @@ def overlap_sort(
             for freq_id in range(f0_ind + step, last_ind, step):
                 # Get next frequency to sort
                 data_to_sort = self._isel(f=[freq_id])
+                data_to_sort = data_to_sort._assign_coords(f=[self.monitor.freqs[f0_ind]])
```

Comment:
I am confused, why is this needed? For a regular DataArray, isel-ing already redefines the coordinates, too. Is something not working right in our `_isel` wrapper?

![image](https://github.com/flexcompute/tidy3d/assets/92756559/9cb10443-416b-42f1-8446-3258332e55d2)

Reply:
It's a bit of a hack -- I want to isel `f=[freq_id]`, which is the next frequency to sort. But, I want to compute outer_dot against the base frequency `f0`. But because of how outer_dot works, it only keeps frequencies which are common between the two datasets. So, here, I isel `f=[freq_id]`, then I assign coords to `f0`, so that outer_dot will think they are at the same frequency and know to compare them.

Reply:
Ohhhhh. Funny. One thing we could do also do is to allow `outer_dot` to work without a frequency dimension, and just pop it? But, eh, I think this is fine.